src/gatsby-node.js

```diff
@@ -76,23 +90,75 @@ exports.sourceNodes = async (
     }
 
     if (contentFieldsResult.data.data) {
-      const contentFields = Object.entries(contentFieldsResult.data.data);
-      contentFields.forEach(([key, value]) => {
-        const gatsbyContentField = Object.assign({}, key, value, {
+
+      const entries = Object.entries(contentFieldsResult.data.data)
+
+      let allCollection = [],
+        allfields = [];
+
+      //Extract Collections and Content Fields into separate variables
+      entries.forEach(([key, value]) => {
+        if (typeof (value) == "object") {
```

Comment:
For a collection, `value` will be an array so this comparison doesn't seem correct.